browser/policy/brave_origin_policy_provider.cc

```diff
@@ -0,0 +1,216 @@
+  SetBraveOriginPolicyWithPreference(
+      policy_map, policy_settings, policy::key::kTorDisabled,
+      tor::prefs::kTorDisabled, base::Value(true));
+#endif
+  SetBraveOriginPolicyWithPreference(
+      policy_map, policy_settings, policy::key::kBraveStatsPingEnabled,
+      kStatsReportingEnabled, base::Value(false));
+  SetBraveOriginPolicyWithPreference(policy_map, policy_settings,
+                                     policy::key::kBraveP3ADisabled,
+                                     p3a::kP3AEnabled, base::Value(false));
+  SetBraveOriginPolicyWithPreference(
+      policy_map, policy_settings, policy::key::kBraveRewardsDisabled,
+      brave_rewards::prefs::kDisabledByPolicy, base::Value(true));
+  SetBraveOriginPolicyWithPreference(
+      policy_map, policy_settings, policy::key::kBraveWalletDisabled,
+      brave_wallet::prefs::kDisabledByPolicy, base::Value(true));
+  SetBraveOriginPolicyWithPreference(
+      policy_map, policy_settings, policy::key::kBraveAIChatEnabled,
+      ai_chat::prefs::kEnabledByPolicy, base::Value(false));
+#if BUILDFLAG(ENABLE_SPEEDREADER)
+  SetBraveOriginPolicyWithPreference(
+      policy_map, policy_settings, policy::key::kBraveSpeedreaderDisabled,
+      speedreader::kSpeedreaderDisabledByPolicy, base::Value(true));
+#endif
+  SetBraveOriginPolicyWithPreference(
+      policy_map, policy_settings, policy::key::kBraveNewsDisabled,
+      brave_news::prefs::kBraveNewsDisabledByPolicy, base::Value(true));
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+  SetBraveOriginPolicyWithPreference(
+      policy_map, policy_settings, policy::key::kBraveVPNDisabled,
+      brave_vpn::prefs::kManagedBraveVPNDisabled, base::Value(true));
+#endif
+  SetBraveOriginPolicyWithPreference(
+      policy_map, policy_settings, policy::key::kBraveTalkDisabled,
+      kBraveTalkDisabledByPolicy, base::Value(true));
+
+  return bundle;
+}
+
+bool BraveOriginPolicyProvider::IsPolicySetByExternalProvider(
+    const std::string& policy_key) const {
+  // We need to check the global policy service to see if this policy
+  // is already set by other providers (registry, plist files, etc.)
+
+  if (!policy_service_) {
+    return false;
+  }
+
+  // Check if the policy is set by any provider
+  const policy::PolicyMap& policies = policy_service_->GetPolicies(
+      policy::PolicyNamespace(policy::POLICY_DOMAIN_CHROME, std::string()));
+
+  const policy::PolicyMap::Entry* entry = policies.Get(policy_key);
+  if (!entry) {
+    return false;
+  }
+
+  // Check if the source is something other than our BraveOrigin source
+  // If it's set by registry, plist, etc., we should respect that
+  return entry->source != policy::POLICY_SOURCE_ENTERPRISE_DEFAULT;
+}
+
+void BraveOriginPolicyProvider::SetBraveOriginPolicyWithPreference(
+    policy::PolicyMap& policy_map,
+    const base::Value::Dict& policy_settings,
+    const std::string& policy_key,
+    const std::string& pref_name,
+    const base::Value& default_value) {
```

Comment:
If we do choose to specify the default values in this function, should we change this to `const bool` to remove the `base::Value` constructor calls at the callsites?